client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -54,15 +54,26 @@ const PatternAssembler: Step = ( { navigation } ) => {
 		return pageTemplate;
 	};
 
+	let incrementIndex = 0;
 	const addSection = ( pattern: Pattern ) => {
+		incrementIndex++;
 		if ( sectionPosition !== null ) {
 			setSections( [
 				...sections.slice( 0, sectionPosition ),
-				pattern,
+				{
+					...pattern,
+					key: `${ incrementIndex }-${ pattern.id }`,
```

Comment:
I think we could reuse the id of the existing section in current position

Reply:
Good catch!

Reply:
Sorry I misunderstood your tip, it should be \`key: sections\[ sectionPosition \].key,\`